docs/path-resolution.rst

```diff
@@ -0,0 +1,690 @@
+    disk when compiling on Windows but not on other platforms:
+
+    .. code-block:: solidity
+
+        import "C:\\project\\lib\\token.sol"; // source unit name: C:\project\lib\token.sol
+
+    To compile such a project on a different platform you would have to use the Standard JSON
+    interface and provide the source directly under the right source unit name.
+    For this reason relying on platform-specific behaviour of the file loader is highly discouraged.
+
+.. index:: ! relative import, ! import; relative
+.. _relative-imports:
+
+Relative Imports
+================
+
+An import starting with ``./`` or ``../`` is a *relative import*.
+Such imports specify the path relative to the source unit name of the importing source unit:
+
+.. code-block:: solidity
+    :caption: /project/lib/math.sol
+
+    import "./util.sol" as util;    // source unit name: /project/lib/util.sol
+    import "../token.sol" as token; // source unit name: /project/token.sol
+
+.. code-block:: solidity
+    :caption: lib/math.sol
+
+    import "./util.sol" as util;    // source unit name: lib/util.sol
+    import "../token.sol" as token; // source unit name: token.sol
+
+.. note::
+
+    Do not confuse relative imports with relative paths.
+    Both ``util.sol`` and ``./util.sol`` specify relative paths on disk but these paths are treated
+    very differently when used in imports.
+    Only the latter creates a relative import.
+
+    Consider the following example:
+
+    .. code-block:: solidity
+       :caption: /project/lib/math.sol
+
+       import "/project/lib/util.sol" as util1; // source unit name: /project/lib/util.sol
+       import "./util.sol" as util2;            // source unit name: /project/lib/util.sol
+       import "util.sol" as util3;              // source unit name: util.sol
+
+    In the situation above the first and the second import are equivalent and refer to the same
+    source unit in the virtual filesystem.
+    The compiler will recognize that the source has already been compiled when it encounters
+    ``./util.sol`` and will not try to compile it again.
+    This is not the case with the third import.
+    When asked for ``util.sol`` with a direct import, the compiler will try to find exactly that.
+    The entry with the source unit name of ``/project/lib/util.sol`` will not be used.
+
+    Even if you run the compiler from within ``/project/lib/`` the relative ``util.sol`` will only
+    get resolved into ``/project/lib/util.sol`` by the file loader.
+    When the loader returns the source, the compiler will compile it but still place it under
+    ``util.sol`` and not ``/project/lib/util.sol`` in the virtual filesystem.
+
+Unlike in direct imports, the paths used in relative imports do get normalized.
+The normalization rules are the same as for UNIX paths, namely:
+
+- All the ``./`` segments are removed.
+- Every ``../``  segment backtracks one level up in the hierarchy.
+- Multiple slashes are squashed into a single one.
```

Comment:
Is this done to the full path? Maybe we should change that and only modify the initial segment.

Reply:
Only the part that's in the import statement. The parent path it's combined with is not normalized. This is explained in a note later in the text.

Generally in this rewrite I tried to keep it simple in the initial explanations and only went into detail later in the text or in note/warning boxes.